debian/screenruler.1
.TH SCREENRULER 1 "May 4, 2008"

.SH NAME
screenruler \- a graphical ruler to measure objects on screen

.SH SYNOPSIS
\fBscreenruler\fP

.SH DESCRIPTION
\fBScreenRuler\fP is a simple graphical ruler which allows you to measure
horizontal or vertically any object on screen using different metric
systems (pixels, centimetres, inches, picas, points or as a percentage).
.PP
It was created for GNOME desktops and can be controled either using the
mouse or, if you need more precision, with the keyboard. If you need it,
it also has an option to keep on top of other windows.

.SH OPTIONS
As already said, \fBScreenRuler\fP is a graphical application, and as such it
takes no command line options. All configuration must be done using its
GTK interface.

.SH AUTHORS
\fBScreenRuler\fP was created by Ian McIntosh <ian@openanswers.org> and this
manual page was written by Siegfried-A. Gevatter <rainct@ubuntu.com>.

.SH LICENSE
\fBScreenRuler\fP has been released under the GNU General Public License,
version 2 or later.


debian/changelog
screenruler (0.960+bzr41+deb10-1) unstable; urgency=medium

  * added myself as uploader
  * recompiled for debian10
  * upgraded dh level to 11, Standards-Version to 4.3.0
  * Closes: #931362

 -- Georges Khaznadar <georgesk@debian.org>  Wed, 03 Jul 2019 08:38:10 +0200

screenruler (0.960+bzr41-1.2) unstable; urgency=medium

  * Non-maintainer upload.
  * Apply patch for Ruby 1.9 support, drop hardcoding of 1.8 (Closes: #734954)

 -- Christian Hofstaedtler <zeha@debian.org>  Thu, 16 Jan 2014 11:08:20 +0100

screenruler (0.960+bzr41-1.1) unstable; urgency=low

  * Non-maintainer upload.
  * debian/control: Replace obsolete libgettext-ruby1.8 dependency with
    ruby-gettext. (Closes: #709336)

 -- Felix Geyer <fgeyer@debian.org>  Mon, 03 Jun 2013 18:20:27 +0200

screenruler (0.960+bzr41-1) unstable; urgency=low

  * New upstream version + modifications from trunk:
     - Port to GtkBuilder and replace GConf with a YAML file (Closes: #639437,
       #654710).
     - Handle symlinks in screenruler.rb.
     - Added a help window and internationalization support.
     - Various bug fixes.
  * debian/control:
     - Update dependencies.
  * debian/screenruler.xpm:
     - Fix typo in Catalan translation and add a German translation.
  * debian/watch:
     - Delete it (no point, upstream version numbers are inconsistent).
  * screenruler.rb:
     - Hardcode Ruby 1.8.
  * Change to source format "3.0 (quilt)" and debhelper compatibility
    version 7, switch from CDBS to dh and bump Standards-Version to 3.9.2.

 -- Siegfried-Angel Gevatter Pujals <rainct@ubuntu.com>  Tue, 24 Jan 2012 23:43:17 +0100

screenruler (0.900+bzr27-1) experimental; urgency=low

  * New upstream version + modifications from trunk:
     - Improved the "Track Mouse" feature.
  * debian/control:
     - Set libgtk2-ruby dependency minimum version to 0.18.1 (Closes: #508788).
     - Bump Standards-Version to 3.9.1.
  * Update copyright years and debian/install.
  * Add debian/source/format with content "1.0".

 -- Siegfried-Angel Gevatter Pujals <rainct@ubuntu.com>  Sun, 05 Dec 2010 14:13:41 +0100

screenruler (0.891+bzr25-1) unstable; urgency=low

  The "waiting for a sponsor since a year" release.

  * New upstream version + modifications from trunk:
     - Ask GDK for DPI settings, and allow users to override them in
       the configuration dialogue (Closes: #509049) (LP: #309159).
     - All patches except for hardcode-data-path.patch are upstreamed.
  * debian/{control,watch,copyright}:
     - Launchpad project name changed from "screen-ruler" to "screenruler"
       (Closes: #550767).
  * debian/copyright:
     - Change it to the machine readable format.
  * debian/screenruler.xpm:
     - Change it to the new tango-style image.
  * debian/{rules,install,links}:
     - Install screenruler.rb into /usr/share/screenruler and symlink it
       from there into /usr/bin.

 -- Siegfried-Angel Gevatter Pujals <rainct@ubuntu.com>  Tue, 01 Dec 2009 18:39:20 +0100

screenruler (0.85-2) unstable; urgency=low

  * debian/patches/fix-color-parsing.patch:
     - The colour selection is not correctly saved, and if you change th
       colour screenruler may afterwards fails to start. The problem was
       that the string was being parsed as 8-bit, but it's 16-bit (LP: #273684).
  * debian/control:
     - Change the Maintainer field so that it doesn't contain my nick anymore.

 -- Siegfried-Angel Gevatter Pujals <rainct@ubuntu.com>  Fri, 19 Dec 2008 17:28:20 +0100

screenruler (0.85-1) unstable; urgency=low

  * Initial release (Closes: #466691).
  * debian/patches/change-icon-path.patch:
     - Patch screenruler.glade so that it looks in /usr/share/pixmaps for
       the application's icon.
  * debian/patches/hardcode-data-path.patch:
     - Hardcode the path to /usr/share/screenruler so that the executable
       finds it.

 -- Siegfried-Angel Gevatter Pujals (RainCT) <rainct@ubuntu.com>  Tue, 04 Mar 2008 16:08:02 +0100


screenruler.rb
#!/usr/bin/env ruby
#coding: utf-8

 ###############################################################################
 #  Copyright 2011 Ian McIntosh <ian@openanswers.org>
 #
 #  This program is free software; you can redistribute it and/or modify
 #  it under the terms of the GNU General Public License as published by
 #  the Free Software Foundation; either version 2 of the License, or
 #  (at your option) any later version.
 #
 #  This program is distributed in the hope that it will be useful,
 #  but WITHOUT ANY WARRANTY; without even the implied warranty of
 #  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 #  GNU Library General Public License for more details.
 #
 #  You should have received a copy of the GNU General Public License
 #  along with this program; if not, write to the Free Software
 #  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
 ###############################################################################

Dir.chdir(File.dirname(File.expand_path(File.symlink?(__FILE__) ? File.readlink(__FILE__) : __FILE__)))		# So that this file can be run from anywhere
$LOAD_PATH << './utils'

require 'gettext'		# Internationalization Support
include GetText
bindtextdomain("screenruler", :path => "locale")

###################################################################
# Constants
###################################################################
UNIT_PIXELS, UNIT_CENTIMETERS, UNIT_INCHES, UNIT_PICAS, UNIT_POINTS, UNIT_PERCENTAGE = (0..5).to_a
UNIT_LAST = UNIT_PERCENTAGE

APP_NAME					= _('Screen Ruler')
APP_COPYRIGHT			= "Copyright (c) #{Time.now.year} Ian McIntosh"
APP_AUTHORS 			= ['Ian McIntosh <ian@openanswers.org>']
APP_ARTISTS				= ['János Horváth <horvathhans@gmail.com>']
APP_VERSION				= '0.9.6'
APP_LOGO_FILENAME = 'screenruler-logo.png'

SETTINGS_SUBDIRECTORY_NAME = 'screenruler'
SETTINGS_FILE_NAME = 'settings.yml'

###################################################################
# Includes
###################################################################
puts _('Loading libraries...')

require 'addons_ruby'									# for multi-file 'require'
require 'gtk2', 'settings', 'addons_gtk', 'ruler_window', 'preferences_window', 'help_window'

###################################################################
# Main
###################################################################
Gtk.init

APP_ICON_LIST = ['screenruler-icon-16x16.png', 'screenruler-icon-32x32.png', 'screenruler-icon-64x64.png'].collect { |filename| Gdk::Pixbuf.new(filename) }

#
# Load Settings
#
settings_directory = File.join(GLib.user_config_dir, SETTINGS_SUBDIRECTORY_NAME)
Dir.mkdir(GLib.user_config_dir) rescue nil ; Dir.mkdir(settings_directory) rescue nil
settings_file_path = File.join(settings_directory, SETTINGS_FILE_NAME)
settings = Settings.new.load(settings_file_path)

#
# Startup
#
puts _('Creating windows...')
	$preferences_window = PreferencesWindow.new
	$ruler_window = RulerWindow.new
	$ruler_popup_menu = RulerPopupMenu.new
	$help_window = HelpWindow.new

puts _('Reading settings...')
	$preferences_window.read_settings(settings)
	$ruler_window.read_settings(settings)
	$ruler_popup_menu.read_settings(settings)

puts _('Presenting ruler...')
	$ruler_window.present

begin
	Gtk.main
ensure
	puts _('Shutting down...')
	[$ruler_window, $ruler_popup_menu, $preferences_window].each { |win| win.hide }				# feels snappy
	[$ruler_window, $ruler_popup_menu, $preferences_window].each { |win| win.write_settings(settings) }
	settings.save(settings_file_path)
end

# Local Variables:
# tab-width: 2
# End:
